java/spark-1.5.2/backend.scala

```diff
@@ -121,6 +122,10 @@ class Backend() {
     isBatch = isBatchParam
   }
 
+  def setPreCommandHooks(preCommandHooksParam: List[Runnable]): Unit = {
+    preCommandHooks = preCommandHooksParam
```

Comment:
```suggestion
    this.preCommandHooks = preCommandHooksParam
```

Reply:
Also why not doing the wrapping in `Option` right here and just keep `preCommandHooksOption`

Reply:
Discussed offline, but the `Runnable` I pass downstream needs to reference a variable local to `Backend`, so that I can dynamically modify it.  It's possible to define both `preCommandHooks` and `preCommandHooksOption` as instance variables, but I thought it would be cleaner to just construct it before I need to use it.